ui/appui-react/src/appui-react/preview/PreviewFeatures.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+// Defined as an object so we can access the keys for runtime validation.
+
+import { useSelector } from "react-redux";
+import type { FrameworkRootState } from "../redux/StateManager";
+import { UiFramework } from "../UiFramework";
+
+/**
+ * List of known preview features.
+ */
+const knownFeaturesObject = {
+  /**
+   * If true, the panels and tool settings will always be rendered over the content.
+   * The content will never change size.
+   */
+  contentAlwaysMaxSize: true as boolean, // So `typeof` below do not type this `true` but `boolean`
+};
+
+/**
+ * List of preview features that can be enabled/disabled.
+ * This list is expected to change over time, the interface is made
+ * so that new features can be added or removed without breaking existing code.
+ * A console warning will simply appear if unknown features are passed.
+ * @alpha
+ */
+export interface PreviewFeatures extends Partial<typeof knownFeaturesObject> {
```

Comment:
No, I had that initially but this causes us to have to edit 3 times everytime we want to add/remove a feature, and end up with 3 sources of truth, that's why I changed it for this single object.

Reply:
It should be two type-safe edits only (i.e. `knownFeaturesObject` will complain after a new property is added to `KnownPreviewFeatures`). The source of truth is `KnownPreviewFeatures`.

Reply:
The way I had done it before was with an Array of keys directly, which didnt cause the type to complain. I want to reduce the boilerplate as much as I can, but I admit that if the type is an elaborate object, it will require multiple edits anyway, or be really complex, so let's go with this approach instead.